actpool/actqueue.go

```diff
@@ -103,17 +102,16 @@ func NewActQueue(ap *actPool, address string, ops ...ActQueueOption) ActQueue {
 	return aq
 }
 
-// Overlap returns whether the current queue contains the given nonce
-func (q *actQueue) Overlaps(act action.SealedEnvelope) bool {
-	_, exist := q.items[act.Nonce()]
-	return exist
-}
-
 // Put inserts a new action into the map, also updating the queue's nonce index
 func (q *actQueue) Put(act action.SealedEnvelope) error {
 	nonce := act.Nonce()
-	if _, exist := q.items[nonce]; exist {
-		return errors.Wrap(action.ErrNonce, "duplicate nonce")
+	if actInPool, exist := q.items[nonce]; exist {
+		// act of higher gas price cut in line
+		if act.GasPrice().Cmp(actInPool.GasPrice()) != 1 {
+			return errors.Wrap(action.ErrNonce, "gas price is smaller than the act of same nonce")
+		}
+		q.items[nonce] = act
```

Comment:
simply replace the action of the same nonce works fine in this case, just because https://github.com/iotexproject/iotex-core/blob/976c94abd634c2a7f2a998aacd776d94db8b5c83/actpool/actqueue.go#L32 uses nonce for comparison. Moreover, the ttl of the new action will be the same as the old one.

We could first delete the old action, removing it from the items and index, and then re-add it into the queue